lib/matplotlib/scale.py

```diff
@@ -478,10 +478,110 @@ def get_transform(self):
         return self._transform
 
 
+def _mask_non_logit(a):
+    """
+    Return a Numpy masked array where all values outside ]0, 1[ are
+    masked. If all values are inside ]0, 1[, the original array is
+    returned.
+    """
+    mask = (a <= 0.0) | (a >= 1.0)
+    if mask.any():
+        return ma.MaskedArray(a, mask=mask)
+    return a
+
+
+def _clip_non_logit(a):
```

Comment:
You should make a copy of the input array either here or where `_handle_non_affine` is called.  Modifying user objects is bad practice.